src/coreclr/scripts/superpmi.py

```diff
@@ -1686,6 +1686,10 @@ async def create_one_artifact(jit_path: str, location: str, flags) -> str:
                     if current_text_diff is not None:
                         logging.info("Textual differences found in generated asm.")
 
+                        asm_diffs_location = os.path.join(asm_root_dir, "asm_diffs.diff")
+                        os.system("git diff --no-index %s %s > %s" % (base_asm_location, diff_asm_location, asm_diffs_location))
+                        logging.info("  Creating git diff file at %s", asm_diffs_location)
```

Comment:
You should use subprocess for this, check the other places in the file that invokes `git` for the right way to do it.
Also this might create a massive file and take a very long time for large diffs, so it should probably be optional.

Reply:
Will look at using the subprocess.

I'm not convinced about the performance concerns though considering we already upload the full "base" and "diffs" folders and run "jit-analyze" against it.

Reply:
We actually only upload the top N diff files IIRC, cc @kunalspathak.
Have you measured how long the git diff takes for two large changes? Maybe @kunalspathak has an easy way to change RA that will result in a lot of diffs so that we can test.

Reply:
Does the current code condition this on being CI-only? I would be concerned about the performance of local runs, in my experience the slowest part of diffing is the IO-bound writing out of `.dasm`s.

Reply:
> This allows us to quickly look at the asm diffs.

I believe you are planning to add this to be used manually and not in CI, yes? Having an optional flag would be good to have IMO.

> I'm not convinced about the performance concerns though considering we already upload the full "base" and "diffs" folders and run "jit-analyze" against it.

We only upload the diff files. Here is what it does on helix:
1. Run superpmi asmdiff produces the before/after raw bytes and do the comparison. It tracks all the methods whose raw bytes are different.
2. Run superpmi replay for before/after with `JitDisasm=*` for all the methods that were different in step 1.
3. Run `jit-analyze` and produces the result that you see under "Extensions"
4. Uploads just top 20 or so files from base/diff directories to get a sense of what diffs are.

Reply:
I guess this is with the arg "--retainOnlyTopFiles".

Reply:
correct.

Reply:
I can run the diff after jit-analyze since jit-analyze will delete files.

Reply:
Well, it depends on if you want to run during CI or manually during local testing. `retainOnly` is only passed during CI runs https://github.com/dotnet/runtime/blob/15dda0efe52c2aaa7a4e783f443eefb558529248/src/coreclr/scripts/superpmi_diffs.py#L220 

So if this just for local runs, it doesn't matter when you run.

Reply:
I added a `--git_diff` option. It's not enabled by default.

Reply:
Ideally, I would want this to run in CI because reviewers could just download the diff to see what has changed.